roles/nvidia_smi/tasks/main.yml

```diff
@@ -0,0 +1,21 @@
+---
+
+- name: check for nvidia_smi
+  stat:
+    path: '{{ nvidia_smi_path }}nvidia-smi'
+  register: nvidia_smi_path
```

Comment:
Are you overloading the same var? 

I would recommend `nvidia_path` and `nvidia_binary` and have `when: nvidia_binary.stat.exists`

Reply:
Do you mean:

```
nvidia_path: '/usr/bin/'
nvidia_binary: '/usr/bin/nvidia-smi'
```
??

Reply:
Well. I am recommending to not overload `nvidia_smi_path` var. Between the user provided path + the register.

Reply:
I see now. I'll remove the register statement since we already have the var